react-components/src/data-providers/core-dm-provider/filterNodesByMappedTo3d.ts

```diff
@@ -61,21 +67,79 @@ export async function filterNodesByMappedTo3d(
 
   const data = await Promise.all(result);
 
-  return data.filter(isDefined);
+  return data
+    .map((viewWithNodes) => ({
+      view: transformViewItemToSource(viewWithNodes.view),
+      instances: viewWithNodes.instances
+    }))
+    .filter(isDefined);
+}
+
+type SelectSourcesType = [
+  { source: typeof COGNITE_3D_OBJECT_SOURCE; properties: Cognite3DObjectProperties },
+  { source: typeof COGNITE_CAD_NODE_SOURCE; properties: { object3D: DmsUniqueIdentifier } },
+  {
+    source: typeof COGNITE_POINT_CLOUD_VOLUME_SOURCE;
+    properties: { object3D: DmsUniqueIdentifier };
+  },
+  {
+    source: typeof COGNITE_IMAGE_360_SOURCE;
+    properties: { collection360: DmsUniqueIdentifier };
+  }
+];
+
+async function getObject3dsConnectedToAssetAndModelRevisions(
+  nodes: InstancesWithViewDefinition[],
+  revisionRefs: DmsUniqueIdentifier[],
+  fdmSdk: FdmSDK
+): Promise<Set<FdmKey>> {
+  const initialIds = nodes.flatMap((node) => node.instances.map(restrictToDmsId));
+
+  const directlyConnectedIds = nodes
+    .flatMap((node) =>
+      node.instances.flatMap((instance) =>
+        getCogniteAssetDirectRelationProperties(instance, node.view)
+      )
+    )
+    .filter(isDefined);
+
+  const mergedIds = concat(initialIds, directlyConnectedIds);
+  const uniqueIds = uniqBy(mergedIds, (id) => `${id.externalId}-${id.space}`);
```

Comment:
Use `createFdmKey` here: `uniqBy(mergedIds, createFdmKey)`

Reply:
Fixed in [69ce933](https://github.com/cognitedata/reveal/pull/5078/commits/69ce933edab88119bc93fe3cfdfa5d811ff3f83f)